drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioSimulationServiceImpl.java

```diff
@@ -196,7 +204,19 @@ public ScenarioSimulationModel load(final Path path) {
         try {
             final String content = ioService.readAllString(Paths.convert(path));
 
-            return ScenarioSimulationXMLPersistence.getInstance().unmarshal(content);
+            ScenarioSimulationModel scenarioSimulationModel = unmarshalInternal(content);
+            Simulation simulation = scenarioSimulationModel.getSimulation();
```

Comment:
@danielezonca 
So, this looks like it is possible and valid to have a **ScenarioSimulationModel** _without_ a **Simulation**: is it true?

Reply:
@gitgabrio 
I think you are referring to the next line 
`if(simulation != null...`
I don't think it should be possible but I added `simulation != null` in if condition just to be sure to not introduce regressions. The old code had no access to `Simulation` so the new method has the same behavior in case of `simulation == null`

Reply:
@danielezonca 
OK, my doubt is: *if* we should not have simulation == null, and we are already checking that, would not be better to explicitly throw an Exception, to make it clear (reading the code) that simulation must not be null?

Reply:
@gitgabrio 
I understand your doubt and I agree with the approach to highlight unexpected situation instead of silent them.
But my point is different: _if_ this can happen and it needs to be managed, it is already managed outside/somewhere else because I'm updating an existing method. This method loads the file and should never fail in a not managed way otherwise this prevents the user the possibility to fix something manually (i.e. dmn file path is wrong).
The meaning of this code inside the if is _optional:_ try to load additional information otherwise go on.